src/libs/actions/Report.js

```diff
@@ -401,6 +440,9 @@ function fetchChatReportsByIDs(chatList, shouldRedirectIfInaccessible = false) {
 
             // Fetch the personal details if there are any
             PersonalDetails.getFromReportParticipants(_.values(simplifiedReports));
+
+            // Configure Report Name and Report Icon
+            configureReportNameAndIcon(_.values(simplifiedReports));
```

Comment:
There could be race conditions here personal details may have not been stored in Onyx when this is called.

Reply:
I think calling configureReportNameAndIcon() from PersonalDetails.getFromReportParticipants() might be the solution. If  we do so, we will not require to connect Onyx for personal details later. It will solve both the issues you raised.

Reply:
```
            Promise.resolve(PersonalDetails.getFromReportParticipants(_.values(simplifiedReports))).then(
                () => configureReportNameAndIcon(_.values(simplifiedReports)),
            );
```
@parasharrajat This should solve the issue.

Reply:
```
            Promise.resolve(PersonalDetails.getFromReportParticipants(_.values(simplifiedReports))).then(
                details => configureReportNameAndIcon(_.values(simplifiedReports), details),
            );
```
@parasharrajat It will solve the issue. Any suggestion?

Reply:
It won't be needed when `PersonalDetails.getFromReportParticipants` will return a promise.